src/Time/Bool.hs

```diff
@@ -1,39 +1,38 @@
-{-# LANGUAGE AllowAmbiguousTypes        #-}
-{-# LANGUAGE ConstraintKinds            #-}
-{-# LANGUAGE CPP                        #-}
-{-# LANGUAGE DataKinds                  #-}
-{-# LANGUAGE DeriveGeneric              #-}
-{-# LANGUAGE ExplicitForAll             #-}
-{-# LANGUAGE FlexibleContexts           #-}
-{-# LANGUAGE InstanceSigs               #-}
-{-# LANGUAGE GeneralizedNewtypeDeriving #-}
-{-# LANGUAGE ScopedTypeVariables        #-}
-{-# LANGUAGE TypeApplications           #-}
-{-# LANGUAGE TypeFamilies               #-}
 {-# LANGUAGE TypeInType                 #-}
 {-# LANGUAGE TypeOperators              #-}
 {-# LANGUAGE UndecidableInstances       #-}
 
 module Time.Bool
-       ( And
-       , If
-       , type (==%)
+       ( If
+       , type (&&)
+       , type (==)
        ) where
 
-import Data.Type.Equality (type (==))
+import GHC.TypeNats (Nat)
+import Data.Kind (Type)
 
 import Time.Rational (type (:%), Rat)
 
+import qualified Data.Type.Equality as Equality (type (==))
+
 -- | Type-level 'if'.
 type family If (x :: Bool) a b where
     If 'True  a _ = a
     If 'False _ b = b
 
 -- | Type-level 'and'.
-type family And (a :: Bool) (b :: Bool) where
-  And 'False _ = 'False
-  And _      x = x
+type family (a :: Bool) && (b :: Bool) :: Bool where
+    'False && _ = 'False
+    _      && x = x
+
+-- | The result kind of overloaded equality.
+type family EqK (t :: Type) :: Type
+
+type instance EqK Nat = Bool
+type instance EqK Rat = Bool
+
+-- | Overloaded equality.
+type family (a :: t) == (b :: t) :: EqK t
 
--- | A type family to compute equality of 'Rat's.
-type family (==%) (m :: Rat) (n :: Rat) where
-    (a :% b) ==% (c :% d) = (a == c) `And` (b == d)
+type instance (a :: Nat) == (b :: Nat) = a Equality.== b
+type instance (a :% b)   == (c :% d)   = (a == c) && (b == d)
```

Comment:
Wait a second, so we're using structural equality here? Can you check please, perhaps `==` from `base` works out of the box?

Reply:
Ooh, sorry, you're totally right! We don't need to do anything, everything is already working..